### PR TITLE
Migrate from `IntKey` to new naked int key

### PR DESCRIPTION
**Pull request by ueco-jb**
closes https://github.com/CosmWasm/cw-plus/issues/549

I succesfully replaced every `IntKey` usage both in contracts and packages (other then implementation itself + tests + de).

Looking at overall progress, I could actually remove whole `IntKey` implementation along with deserialization.
I wonder if it wouldn't be too drastic? This seems to be major change, although from end user POV nothing changes - it's all internal.

### CODE REVIEW

**Review by maurolacy — APPROVED** [review]
LGTM. Mostly cosmetic, looks better.

**Comment by maurolacy** [comment]
Regarding deprecation / removal of the `IntKey` types, that would be a breaking change. What we can do is to add a deprecation notice for a while, and the remove, detailing the easy migration path / an example in the docs.

File: packages/storage-plus/src/keys.rs [hunk 1]
```diff
@@ -336,6 +339,7 @@ pub type I128Key = IntKey<i128>;
 ///   let k = U64Key::new(12345);
 ///   let k = U32Key::from(12345);
 ///   let k: U16Key = 12345.into();
+#[deprecated(note = "It is suggested to use naked int types instead of IntKey wrapper")]
```
Comment (maurolacy) [inline packages/storage-plus/src/keys.rs:342]: 👍🏼

File: packages/storage-plus/src/map.rs [hunk 2]
```diff
@@ -562,7 +555,7 @@ mod test {
         let all: StdResult<Vec<_>> = PEOPLE_ID
             .range_de(
                 &store,
-                Some(Bound::Inclusive(U32Key::new(56).into())),
+                Some(Bound::Inclusive(56u32.to_be_bytes().to_vec())),
```
[the remark this answers is not in the document]
Comment (ueco-jb) [reply packages/storage-plus/src/map.rs]: `inclusive_int` actually, but yes - thank you.

**Pull request merged**